Bug #210901: Add fully qualified paths to commands in example hook
scripts that get installed as .tmpl files in new repositories.  Use
$REPOS/hooks/ as the path, to encourage repository admins to copy the
hooks, not just use them in place.

--- a/subversion/libsvn_repos/repos.c
+++ b/subversion/libsvn_repos/repos.c
@@ -275,7 +275,7 @@
 
 #define PREWRITTEN_HOOKS_TEXT                                                 \
   "# For more examples and pre-written hooks, see those in"                NL \
-  "# the Subversion repository at"                                         NL \
+  "# /usr/share/subversion/hook-scripts, and in the repository at"         NL \
   "# http://svn.apache.org/repos/asf/subversion/trunk/tools/hook-scripts/ and"        NL \
   "# http://svn.apache.org/repos/asf/subversion/trunk/contrib/hook-scripts/"          NL
 
@@ -354,8 +354,11 @@
 "REPOS=\"$1\""                                                               NL
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
-"commit-allower.pl --repository \"$REPOS\" --user \"$USER\" || exit 1"       NL
-"special-auth-check.py --user \"$USER\" --auth-level 3 || exit 1"            NL
+"# Exit on all errors."                                                      NL
+"set -e"                                                                     NL
+""                                                                           NL
+"\"$REPOS\"/hooks/commit-allower.pl --repository \"$REPOS\" --user \"$USER\"" NL
+"\"$REPOS\"/hooks/special-auth-check.py --user \"$USER\" --auth-level 3"     NL
 ""                                                                           NL
 "# All checks passed, so allow the commit."                                  NL
 "exit 0"                                                                     NL;
@@ -442,10 +445,13 @@
 "$SVNLOOK log -t \"$TXN\" \"$REPOS\" | \\"                                   NL
 "   grep \"[a-zA-Z0-9]\" > /dev/null || exit 1"                              NL
 ""                                                                           NL
+"# Exit on all errors."                                                      NL
+"set -e"                                                                     NL
+""                                                                           NL
 "# Check that the author of this commit has the rights to perform"           NL
 "# the commit on the files and directories being modified."                  NL
-"commit-access-control.pl \"$REPOS\" \"$TXN\" commit-access-control.cfg || exit 1"
-                                                                             NL
+"\"$REPOS\"/hooks/commit-access-control.pl \"$REPOS\" $TXN \\"               NL
+"  \"$REPOS\"/hooks/commit-access-control.cfg"                               NL
 ""                                                                           NL
 "# All checks passed, so allow the commit."                                  NL
 "exit 0"                                                                     NL;
@@ -756,7 +762,7 @@
 "REPOS=\"$1\""                                                               NL
 "REV=\"$2\""                                                                 NL
                                                                              NL
-"mailer.py commit \"$REPOS\" \"$REV\" /path/to/mailer.conf"                  NL;
+"\"$REPOS\"/hooks/mailer.py commit \"$REPOS\" $REV \"$REPOS\"/mailer.conf"   NL;
 
 #undef SCRIPT_NAME
 
@@ -817,7 +823,8 @@
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
 "# Send email to interested parties, let them know a lock was created:"      NL
-"mailer.py lock \"$REPOS\" \"$USER\" /path/to/mailer.conf"                   NL;
+"\"$REPOS\"/hooks/mailer.py lock \\"                                         NL
+"  \"$REPOS\" \"$USER\" \"$REPOS\"/hooks/mailer.conf"                        NL;
 
 #undef SCRIPT_NAME
 
@@ -876,7 +883,8 @@
 "USER=\"$2\""                                                                NL
 ""                                                                           NL
 "# Send email to interested parties, let them know a lock was removed:"      NL
-"mailer.py unlock \"$REPOS\" \"$USER\" /path/to/mailer.conf"                 NL;
+"\"$REPOS\"/hooks/mailer.py unlock \\"                                       NL
+"  \"$REPOS\" \"$USER\" \"$REPOS\"/hooks/mailer.conf"                        NL;
 
 #undef SCRIPT_NAME
 
@@ -941,8 +949,8 @@
 "PROPNAME=\"$4\""                                                            NL
 "ACTION=\"$5\""                                                              NL
 ""                                                                           NL
-"mailer.py propchange2 \"$REPOS\" \"$REV\" \"$USER\" \"$PROPNAME\" "
-"\"$ACTION\" /path/to/mailer.conf"                                           NL;
+"\"$REPOS\"/hooks/mailer.py propchange2 \"$REPOS\" $REV \\"                  NL
+"  \"$USER\" \"$PROPNAME\" \"$ACTION\" \"$REPOS\"/hooks/mailer.conf"         NL;
 
 #undef SCRIPT_NAME
 
